pkg/reconciler/clusteringress/resources/virtual_service.go

```diff
@@ -155,6 +198,16 @@ func dedup(hosts []string) []string {
 	return sets.NewString(hosts...).List()
 }
 
+func retainLocals(hosts []string) []string {
+	retained := []string{}
+	for _, h := range hosts {
+		if strings.HasSuffix(h, ".svc."+network.GetClusterDomainName()) {
```

Comment:
the `".svc."+network.GetClusterDomainName()` string is constant, so should probably be calculated outside of the loop.